compute_tools/src/migration.rs

```diff
@@ -1,100 +1,257 @@
+use std::{collections::HashMap, fmt::Display};
+
 use anyhow::{Context, Result};
-use postgres::Client;
+use compute_api::spec::Database;
+use postgres::{Client, NoTls};
 use tracing::info;
+use url::Url;
+
+use crate::pg_helpers::get_existing_dbs;
+
+pub(crate) enum Migration<'m> {
+    /// Cluster migrations are things like catalog updates, where they can be run in the default
+    /// Postgres database, but affect every database in the cluster.
+    Cluster(&'m str),
+
+    /// Per-database migrations will be run in every database of the cluster. The migration will
+    /// not be marked as completed after it has been run in every database. We will save the
+    /// "postgres" database for last so that we can commit the transaction as applied in the
+    /// neon_migration.migration_id table.
+    ///
+    /// Please be aware of the race condition that exists for this type of migration. At the
+    /// beginning of running the series of migrations, we get the current list of databases.
```

Comment:
Is it the problem given that we use never version of postgres?

Dowe need to apply this fix in every new database in the cluster until forever?
If so, we can create DDL trigger on database creation

Reply:
The problem only exists for databases created before their computes used the latest Postgres minor releases. I need to come up with a SQL query for understanding how many endpoints are affected. Stay tuned!